packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -424,17 +432,10 @@ export function getStylesDeclarations(
 			? rule.key
 			: kebabCase( rule.key );
 
-		let ruleValue = rule.value;
-		if ( typeof ruleValue !== 'string' && ruleValue?.ref ) {
-			const refPath = ruleValue.ref.split( '.' );
-			ruleValue = compileStyleValue(
-				getValueFromObjectPath( tree, refPath )
-			);
-			// Presence of another ref indicates a reference to another dynamic value.
-			// Pointing to another dynamic value is not supported.
-			if ( ! ruleValue || ruleValue?.ref ) {
-				return;
-			}
+		let ruleValue = getResolvedRefValue( rule.value, tree, undefined );
```

Comment:
I wonder if we could also resolve themeURIs here to removing the necessity to do it in [useGlobalStylesOutputWithConfig](https://github.com/WordPress/gutenberg/blob/43e9107e4ea6cb0115ed08af2ce794d2867f7246/packages/block-editor/src/components/global-styles/use-global-styles-output.js#L1402-L1401)

I don't know, I'll test.